components/calendar/src/chinese.rs

```diff
@@ -379,9 +379,14 @@ impl ChineseBased for Chinese {
 
     const EPOCH: RataDie = CHINESE_EPOCH;
 
+    // Unwrap can be used for this function because it is only called internally when
+    // generating dates from a valid year, month, and day
     #[allow(clippy::unwrap_used)]
-    fn new_chinese_based_date(year: i32, month: u8, day: u8) -> Date<Chinese> {
-        Date::try_new_chinese_date(year, month, day).unwrap()
+    fn new_chinese_based_date(year: i32, month: u8, day: u8) -> ChineseBasedDateInner<Chinese> {
+        Date::try_new_chinese_date(year, month, day)
+            .unwrap()
+            .inner
```

Comment:
no, I think we should construct the CBDI _directly_ . We should use `ArithmeticDate::new_unchecked()` so we don't have to call unwrap

Reply:
Done